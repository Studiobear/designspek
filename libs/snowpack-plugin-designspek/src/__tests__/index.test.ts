import path from 'path'
import { promises as fs } from 'fs'
const snowpackPluginDesignspek = require('../index.ts')

describe('snowpack-plugin-mdx', () => {
  it('should compile .svelte file', async () => {
    const filePath = path.join(__dirname, '__fixtures__/Component.svelte')
    const contents = await fs.readFile(filePath, 'utf-8')
    const plugin = snowpackPluginDesignspek({}, {})
    const result = await plugin.load({ contents, filePath })
    expect(result['.js']).toMatchSnapshot('.js')
  })
  it('should find styled function', async () => {
    const filePath = path.join(__dirname, '__fixtures__/Component.svelte')
    const contents = await fs.readFile(filePath, 'utf-8')
    const plugin = snowpackPluginDesignspek({}, {})
    const result = await plugin.load({ contents, filePath })
    const {code} = result.js
    expect(code).toEqual('.js')
  })
})
