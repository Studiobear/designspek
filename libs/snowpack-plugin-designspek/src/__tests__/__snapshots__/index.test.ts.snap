// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snowpack-plugin-mdx should compile .svelte file: .js 1`] = `
Object {
  "code": "/* generated by Svelte v3.29.0 */
import {
	SvelteComponent,
	append,
	attr,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal
} from \\"svelte/internal\\";

import { styled } from \\"@studiobear/designspek\\";

function create_fragment(ctx) {
	let div;
	let h3;

	return {
		c() {
			div = element(\\"div\\");
			h3 = element(\\"h3\\");
			h3.textContent = \\"Designspek\\";
			attr(div, \\"class\\", container);
		},
		m(target, anchor) {
			insert(target, div, anchor);
			append(div, h3);
		},
		p: noop,
		i: noop,
		o: noop,
		d(detaching) {
			if (detaching) detach(div);
		}
	};
}

const container = \\"go2225017453\\";

class Component extends SvelteComponent {
	constructor(options) {
		super();
		init(this, options, null, create_fragment, safe_not_equal, {});
	}
}

export default Component;",
}
`;
